components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1471,25 +1471,57 @@ void RewardsServiceImpl::EnableGreaseLion(const bool enabled) {
   #endif
 }
 
-void RewardsServiceImpl::StopLedger() {
+void RewardsServiceImpl::StopLedger(StopLedgerCallback callback) {
   BLOG(1, "Shutting down ledger process");
   if (!Connected()) {
     BLOG(1, "Ledger process not running");
     Reset();
     return;
   }
 
-  bat_ledger_->Shutdown(
-      base::BindOnce(&RewardsServiceImpl::OnStopLedger, AsWeakPtr()));
+  bat_ledger_->Shutdown(base::BindOnce(
+      &RewardsServiceImpl::OnStopLedgerInternal,
+      AsWeakPtr(),
+      std::move(callback)));
 }
 
-void RewardsServiceImpl::OnStopLedger(const ledger::Result result) {
+void RewardsServiceImpl::OnStopLedgerInternal(
```

Comment:
In order to minimize confusion between `OnStopLedgerInternal` and `OnStopLedger` we might want to name things a bit differently. Maybe the new one should be something along the lines of:

- `OnStopLedgerForCompleteReset`
- `OnCompleteResetStopLedger`
- `OnCompleteResetLedgerStopped`

etc.

Reply:
Great suggestion, I went with the first one - thanks!